### DISCUSSION TITLE
Transform an object into ordered_json

### DISCUSSION THREAD

**Discussion by Tryliom** [nlohmann/json#3707] [Q&A] [answered]
Using nlohmann json I get map of alphabetical ordered objects, I have found that I can have a ordered_json but the example show only how to do it with a json string here: https://json.nlohmann.me/features/object_order/

I have seen that some project of make a macro for it has been abandonned. How can I transform an object (with macro done) into ordered_json ?

**Comment by falbrechtskirchinger** [comment, marked as answer]
You can write the `to_json`/`from_json` functions yourself as documented [here](https://json.nlohmann.me/features/arbitrary_types/). Just replace `nlohmann::json` with `nlohmann::ordered_json` or use a template type instead.

If you want to use a macro, you'll have to wait for the PR (which has not been abandoned) to be merged.

**Reply by Tryliom** [reply]
I thought it worked but when I display my json, it's still alphabetically ordered even if it's ordered correctly with my function `to_json`

**Reply by falbrechtskirchinger** [reply]
Then you're probably converting it to `nlohmann::json` at some point. Can't help you further without seeing the code (at least a small sample to reproduce the issue).

Edit: Demo https://godbolt.org/z/rMcT7odGE

**Reply by Tryliom** [reply]
Yes I was using a map, I have use the ordered_map instead
